src/test/kotlin/cacophony/semantic/TypeCheckerTest.kt

```diff
@@ -1014,11 +1026,11 @@ class TypeCheckerTest {
     fun `error - function declaration type mismatch - types`() {
         val arg1 = Definition.FunctionArgument(lc, "x", testInt)
         val arg2 = Definition.FunctionArgument(lc, "y", testBoolean)
-        val funDef = Definition.FunctionDeclaration(lc, "f", Type.Functional(lc, listOf(Type.Basic(lc, "Boolean"), Type.Basic(lc, "Int")), Type.Basic(lc, "Int")), listOf(arg1, arg2), testInt, intLiteral)
+        val funDef = Definition.FunctionDeclaration(lc, "f", Type.Functional(lc, listOf(testBoolean, testInt), testInt), listOf(arg1, arg2), testInt, intLiteral)
```

Comment:
I intentionally created here new instances, co that reference-wise equality semantic is not violated (this would not be the same instance if it came from the code)

Reply:
Sorry, now it should be fixed